### PR TITLE
[API Server] Static User Configuration

### LINKED ISSUE #447: API Static User Configuration
When installing KubeCarrier the KubeCarrier API server requires an OIDC setup to allow user logins and thus the UI to work.

Requiring an OIDC installation is a big hurdle to try out KubeCarrier. To ease in this initial installation, we should support static user credentials.

**Example Configuration**

```yaml
apiVersion: operator.kubecarrier.io/v1alpha1
kind: KubeCarrier
metadata:
  name: kubecarrier
spec:
  api:
    staticUsers:
      htpasswdSecret:
        name: kubecarrier-users
```

**Acceptance Criteria**
- allows to configure a `.htpasswd` file to be referenced for static users

### PR DESCRIPTION
**Pull request by jiachengxu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #447

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/internal/resources/apiserver/apiserver.go [hunk 1]
```diff
@@ -48,142 +51,220 @@ var k = kustomize.NewDefaultKustomize()
...
+					"--htpasswd-secret-name=$(HTPASSWD_SECRET_NAME)",
+				)
+				containers[i].Env = append(containers[i].Env,
+					corev1.EnvVar{
+						Name:  "HTPASSWD_SECRET_NAME",
+						Value: c.Spec.StaticUsers.HtpasswdSecret.Name,
+					},
+				)
+				supportedAuth = append(supportedAuth, "Htpasswd")
+				for j, env := range containers[i].Env {
```
Comment (nmiculinic) [inline pkg/internal/resources/apiserver/apiserver.go:142]: why not just add this `AuthModeEnv` at the end, once all valid authorization modes have been processed?

Comment (jiachengxu) [reply pkg/internal/resources/apiserver/apiserver.go]: fixed

**Pull request merged**